src/io/gff.rs

```diff
@@ -0,0 +1,301 @@
+}
+
+
+impl Writer<fs::File> {
+    /// Write to a given file path.
+    pub fn to_file<P: AsRef<Path>>(path: P) -> io::Result<Self> {
+        fs::File::create(path).map(Writer::new)
+    }
+}
+
+
+impl<W: io::Write> Writer<W> {
+    /// Write to a given writer.
+    pub fn new(writer: W) -> Self {
+        Writer { inner: csv::Writer::from_writer(writer).delimiter(b'\t').flexible(true) }
+    }
+
+    /// Write a given GFF record.
+    pub fn write(&mut self, record: Record) -> csv::Result<()> {
+        let attributes;
+        if !record.attributes.is_empty() {
+            attributes = record.attributes.iter().map(|(a, b)| format!("{}={}", a, b)).collect::<Vec<_>>().join(";");
+        } else {
+            attributes = "".to_owned();
+        }
+        self.inner.encode((record.seqname, record.source, record.feature_type, record.start, record.end, record.score, record.strand, record.frame, attributes))
+    }
+}
+
+
+/// A GFF record
+#[derive(RustcEncodable)]
+pub struct Record {
+    seqname: String,
+    source: String,
+    feature_type: String,
+    start: u64,
+    end: u64,
+    score: String,
+    strand: String,
+    frame: String,
+    attributes: HashMap<String, String>,
+}
+
+impl Record {
+    /// Create a new GFF record.
+    pub fn new() -> Self {
+        Record {
+            seqname: "".to_owned(),
+            source: "".to_owned(),
+            feature_type: "".to_owned(),
+            start: 0,
+            end: 0,
+            score: ".".to_owned(),
+            strand: ".".to_owned(),
+            frame: "".to_owned(),
+            attributes: HashMap::<String, String>::new(),
+        }
+    }
+
+    /// Sequence name of the feature.
+    pub fn seqname(&self) -> &str {
+        &self.seqname
+    }
+
+    /// Source of the feature.
+    pub fn source(&self) -> &str {
+        &self.source
+    }
+
+    /// Type of the feature.
+    pub fn feature_type(&self) -> &str {
+        &self.feature_type
+    }
+
+    /// Start position of feature (1-based).
+    pub fn start(&self) -> &u64 {
+        &self.start
+    }
+
+    /// End position of feature (1-based, not included).
+    pub fn end(&self) -> &u64 {
+        &self.end
+    }
+
+    /// Score of feature
+    pub fn score(&self) -> Option<u64> {
+        match self.score.as_ref() {
+            "." => None,
+            _ => self.score.parse::<u64>().ok(),
+        }
+    }
+
+    /// Strand of the feature.
+    pub fn strand(&self) -> Option<Strand> {
+        match self.strand.as_ref() {
+            "+" => Some(Strand::Forward),
+            "-" => Some(Strand::Reverse),
+            _ => None,
+        }
+    }
+
+    /// Frame of the feature.
+    pub fn frame(&self) -> &str {
+        &self.frame
+    }
+
+    /// Attribute of feature
+    pub fn attributes(&self) -> &HashMap<String, String> {
+        &self.attributes
+    }
+    
+    /// Get mutable reference on seqname of feature.
+    pub fn seqname_mut(&mut self) -> &mut str {
```

Comment:
You should return `&mut String` here (also in the other cases), because otherwise, people won't be able to clear and push.